### ISSUE TITLE
I have a error  with  CHAN: REORGANIZE:

### ISSUE THREAD

**Issue by littlegirlpppp** [btcsuite/btcd#1194]
sunbodeMBP:bin sunbo$ ./btcd
2018-06-14 14:14:29.253 [INF] BTCD: Version 0.12.0-beta
2018-06-14 14:14:29.253 [INF] BTCD: Loading block database from '/Users/sunbo/Library/Application Support/Btcd/data/mainnet/blocks_ffldb'
2018-06-14 14:14:29.326 [INF] BTCD: Block database loaded
2018-06-14 14:14:29.338 [INF] INDX: cf index is enabled
2018-06-14 14:14:29.338 [INF] CHAN: Loading block index...
2018-06-14 14:14:31.483 [INF] CHAN: Chain state (height 384560, hash 000000000000000005a15c23c9f1d67a8e5d93944e73fb74a4177c0611992d02, totaltx 93547834, work 14942772535223485067948946)
2018-06-14 14:14:31.497 [INF] RPCS: RPC server listening on 127.0.0.1:8334
2018-06-14 14:14:31.497 [INF] RPCS: RPC server listening on [::1]:8334
2018-06-14 14:14:31.504 [INF] AMGR: Loaded 278 addresses from file '/Users/sunbo/Library/Application Support/Btcd/data/mainnet/peers.json'
2018-06-14 14:14:31.506 [INF] CMGR: 40 addresses found from DNS seed seed.bitcoin.sipa.be
2018-06-14 14:14:31.513 [INF] CMGR: Server listening on [::]:8333
2018-06-14 14:14:31.515 [INF] CMGR: 38 addresses found from DNS seed dnsseed.bitcoin.dashjr.org
2018-06-14 14:14:31.515 [INF] CMGR: 38 addresses found from DNS seed seed.bitcoin.jonasschnelli.ch
2018-06-14 14:14:31.517 [INF] CMGR: Server listening on 0.0.0.0:8333
2018-06-14 14:14:31.580 [INF] CMGR: 33 addresses found from DNS seed dnsseed.bluematt.me
2018-06-14 14:14:31.635 [INF] CMGR: 41 addresses found from DNS seed seed.bitcoinstats.com
2018-06-14 14:14:31.676 [INF] CMGR: 40 addresses found from DNS seed seed.bitnodes.io
2018-06-14 14:14:32.527 [INF] SYNC: New valid peer 64.99.224.88:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:32.527 [INF] SYNC: Syncing to block height 527381 from peer 64.99.224.88:8333
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 180.181.118.167:8333 (outbound) (/Satoshi:0.16.0(MY_BALLS)/)
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 73.250.151.156:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 82.95.192.186:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 103.82.56.24:8333 (outbound) (/Satoshi:0.15.99/)
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 155.143.140.186:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:32.528 [INF] SYNC: New valid peer 155.143.140.186:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:32.530 [INF] SYNC: Lost peer 82.95.192.186:8333 (outbound)
2018-06-14 14:14:33.221 [INF] SYNC: New valid peer 75.24.121.227:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:33.244 [INF] SYNC: New valid peer 176.9.158.150:8333 (outbound) (/Satoshi:0.16.0/)
2018-06-14 14:14:35.688 [INF] CHAN: REORGANIZE: Block 0000000000000000063986c60f4b617bf7310b58cdde652ceddd3addc62eb2cc is causing a reorganize.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x135b6d7]

goroutine 41 [running]:
github.com/btcsuite/btcd/blockchain.(*BlockChain).reorganizeChain(0xc420073e00, 0xc428b3a2d0, 0xc428b3a2a0, 0xc4265000d0, 0x1)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/blockchain/chain.go:1009 +0x12c7
github.com/btcsuite/btcd/blockchain.(*BlockChain).connectBestChain(0xc420073e00, 0xc428a0f8f0, 0xc4289fccb0, 0x0, 0x0, 0x0, 0xc429455ad8)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/blockchain/chain.go:1129 +0xa5f
github.com/btcsuite/btcd/blockchain.(*BlockChain).maybeAcceptBlock(0xc420073e00, 0xc4289fccb0, 0xffffffff00000000, 0x0, 0x0, 0x1ac8100)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/blockchain/accept.go:79 +0x207
github.com/btcsuite/btcd/blockchain.(*BlockChain).ProcessBlock(0xc420073e00, 0xc4289fccb0, 0xc400000000, 0xc426980000, 0x0, 0x0)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/blockchain/process.go:228 +0x2b5
github.com/btcsuite/btcd/netsync.(*SyncManager).handleBlockMsg(0xc420162420, 0xc4262c0fa0)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/netsync/manager.go:569 +0x205
github.com/btcsuite/btcd/netsync.(*SyncManager).blockHandler(0xc420162420)
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/netsync/manager.go:1166 +0x548
created by github.com/btcsuite/btcd/netsync.(*SyncManager).Start
	/Users/sunbo/Desktop/PubGo/src/github.com/btcsuite/btcd/netsync/manager.go:1388 +0xe3

**Comment by liyue201** [comment]
i met the same error

**Comment by littlegirlpppp** [comment]
Is it bug?

**Comment by liyue201** [comment]
应该是吧，经常电脑断电后再也同步不了了，删掉数据重新同步

**Comment by littlegirlpppp** [comment]
I have deleted the data twice, but this error has occurred

**Comment by githorray** [comment]
Mine is dead as well:

```
2018-06-21 13:17:16.761 [INF] CHAN: REORGANIZE: Block 0000000000000000010dd18650065dba688100ab1c8807e679ed028adcdc9d5c is causing a reorganize.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x75a9d7]

goroutine 34 [running]:
github.com/btcsuite/btcd/blockchain.(*BlockChain).reorganizeChain(0xc4204c7680, 0xc4251104e0, 0xc4251104b0, 0xc42013c620, 0x1)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/blockchain/chain.go:1009 +0x12c7
github.com/btcsuite/btcd/blockchain.(*BlockChain).connectBestChain(0xc4204c7680, 0xc421fce0e0, 0xc4220be700, 0x0, 0x0, 0x0, 0xc42ade5ad8)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/blockchain/chain.go:1129 +0xa5f
github.com/btcsuite/btcd/blockchain.(*BlockChain).maybeAcceptBlock(0xc4204c7680, 0xc4220be700, 0xffffffff00000000, 0x0, 0x0, 0xec77a0)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/blockchain/accept.go:79 +0x207
github.com/btcsuite/btcd/blockchain.(*BlockChain).ProcessBlock(0xc4204c7680, 0xc4220be700, 0xc400000000, 0xc429e70000, 0x0, 0x0)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/blockchain/process.go:228 +0x2b5
github.com/btcsuite/btcd/netsync.(*SyncManager).handleBlockMsg(0xc4201642c0, 0xc426667e00)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/netsync/manager.go:569 +0x205
github.com/btcsuite/btcd/netsync.(*SyncManager).blockHandler(0xc4201642c0)
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/netsync/manager.go:1166 +0x548
created by github.com/btcsuite/btcd/netsync.(*SyncManager).Start
        /home/ubuntu/gocode/src/github.com/btcsuite/btcd/netsync/manager.go:1388 +0xe3
```

**Comment by qshuai** [comment]
I have synced serval times on many different servers including darwin and ubuntu platform, and I never encounter this error.
the following steps recommend:

- todo check the disk spare space and memory usage.
- todo check the golang version if you btcd binary file is compiled on local machine. 
- [important]todo check if stopping btcd server via `CTRL + C` or `btcctl stop`

**Comment by littlegirlpppp** [comment]
thanks  qshuai:
disk spare space and memory usage is Ok ,the golang version is latest, But  i stoped btcd server via CTRL + C  several times ，So  I will try again

**Comment by liyue201** [comment]
i deleted these lines and fixed it . haha

https://github.com/btcsuite/btcd/blob/86fed781132ac890ee03e906e4ecd5d6fa180c64/blockchain/chain.go#L1009

https://github.com/btcsuite/btcd/blob/86fed781132ac890ee03e906e4ecd5d6fa180c64/blockchain/chain.go#L1012

**Comment by littlegirlpppp** [comment]
what is detachNodes?   If you delete it, are there any other questions

**Comment by davecgh** [comment]
I'll submit a PR to fix this shortly.

**Comment by davecgh** [comment]
This is resolved properly by #1215.
